projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/base/LLVMBasicBlockNode.java

```diff
@@ -55,23 +60,43 @@
     @CompilationFinal private final long[] successorCount;
     @CompilationFinal private long totalExecutionCount = 0;
     private final int blockId;
+    private final String blockName;
+
+    private final BranchProfile controlFlowExceptionProfile = BranchProfile.create();
+
+    @CompilationFinal private SourceSection sourceSection;
 
     @Override
     public void executeVoid(VirtualFrame frame) {
         executeGetSuccessorIndex(frame);
     }
 
-    public LLVMBasicBlockNode(LLVMNode[] statements, LLVMTerminatorNode termInstruction, int blockId) {
+    public LLVMBasicBlockNode(LLVMNode[] statements, LLVMTerminatorNode termInstruction, int blockId, String blockName) {
         this.statements = statements;
         this.termInstruction = termInstruction;
         this.blockId = blockId;
+        this.blockName = blockName;
         successorCount = new long[termInstruction.getSuccessors().length];
     }
 
     @ExplodeLoop
     public int executeGetSuccessorIndex(VirtualFrame frame) {
         for (LLVMNode statement : statements) {
-            statement.executeVoid(frame);
+            try {
+                statement.executeVoid(frame);
+            } catch (ControlFlowException e) {
+                controlFlowExceptionProfile.enter();
```

Comment:
The branch profiling is not absolutely necessary here: Only `abort`, `exit`, and the LLVM IR `unreachable` are `ControlFlowException`s that are mostly not executed. Since they are mostly not executed, they are also not compiled due to the branch profiling in the [basic block node](https://github.com/graalvm/sulong/blob/master/projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/others/LLVMBlockNode.java#L83). I still think that the profiling is useful here, since we might want to add other control flow excecptions in the future.